core/arch/aarch64/proc.c

```diff
@@ -74,6 +74,12 @@ proc_num_simd_saved(void)
     return num_simd_saved;
 }
 
+void
+proc_set_num_simd_saved(int num)
+{
+    num_simd_saved = num;
```

Comment:
Thanks, seen that before. Good point. Unprotecting it for the write seems fine, it will happen only once. With respect to the race question below, I will reply there. But in addition to that, looking at the SELF_UNPROTECT_DATASEC code, it doesn't seem thread-safe. I would expect somebody calling this holding a mutex. But looking at the caller examples in core/, at least some cases don't seem to hold an explicit mutex (maybe they do on a higher level somewhere?). See comment below for more.

Reply:
What about d_r_avx512_code_in_use by the way?

Reply:
Yes, d_r_avx512_code_in_use is the same, though it's x86-only.

In general we did not do a good job porting to ARM: #2502 perhaps covers it.  We were too used to x86's memory model and have stores all over the place that are assumed to be atomic, including some more complex cases explicitly listed in #2502.  Here I would say just use one of the ATOMIC_WRITE or whatnot macros we have to more safely ensure this store is atomic on all (incl non-x86) platforms.

Reply:
Ok. W.r.t. my first comment, I think calling SELF_UNPROTECT_DATASEC/SELF_PROTECT_DATASEC is not subject to a critical race unless somebody changed the protect_mask, which doesn't happen after init I suppose.

Reply:
Done.

Reply:
> looking at the SELF_UNPROTECT_DATASEC code, it doesn't seem thread-safe. I would expect > somebody calling this holding a mutex. 

There is a mutex inside protect_data_section().

Reply:
I see now. Ok.

Reply:
Plus there's a counter, so two competing unprot/prot regions can't result in one of them thinking it's unprotected and crashing on its store.